### PR TITLE
Fix failing toml parser test cases in Windows

### PR DESCRIPTION
**Pull request by parkavi11**
Fixes 
https://github.com/ballerina-platform/ballerina-lang/issues/21638,
https://github.com/ballerina-platform/ballerina-lang/issues/20933,
https://github.com/ballerina-platform/ballerina-lang/issues/20044

## Suggested Labels
Area/BuildTools

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 1]
```diff
@@ -27,11 +27,14 @@
...
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Locale;
 
 /**
  * Test class to populate Manifest object by reading the toml.
  */
 public class ManifestProcessorTest {
+    private static final String OS = System.getProperty("os.name").toLowerCase(Locale.getDefault());
+    boolean flag = true;
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java:37]:
```suggestion
    private boolean flag = true;
```

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 2]
```diff
@@ -126,44 +129,115 @@ public void testLicenseDescription() throws TomlException {
         Assert.assertEquals(manifest.getProject().getLicense().get(1), "Apache-2.0");
     }
 
+    // Disabled on Windows
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java:132]: Shall we change the `description` of the test method appropriately and remove this comment?

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 3]
```diff
@@ -126,44 +129,115 @@ public void testLicenseDescription() throws TomlException {
...
+            Assert.assertEquals(manifest.getDependencies().get(0).getMetadata().getVersion(), "1.1.5");
+            Assert.assertEquals(manifest.getDependencies().get(0).getMetadata().getPath().toString(),
+                    baloPath.toString());
+
+            Files.delete(baloPath);
+            Files.delete(tmpDir);
+        }
     }
 
+
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java:184]: Shall we remove this extra line?

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 4]
```diff
@@ -126,44 +129,115 @@ public void testLicenseDescription() throws TomlException {
...
     @Test(description = "Empty dependency added to the dependencies section has no effect")
     public void testSingleEmptyDependencies() throws TomlException {
         Manifest manifest = ManifestProcessor.parseTomlContentFromString(this.validProjectBlock + "[dependencies] \n " +
                 "string-utils = {} \n");
         Assert.assertEquals(manifest.getDependencies().get(0).getModuleID(), "string-utils");
     }
 
+    // Disabled on Windows
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java:192]: Shall we change the `description` of the test method appropriately and remove this comment?

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java [hunk 5]
```diff
@@ -126,44 +129,115 @@ public void testLicenseDescription() throws TomlException {
...
+            Assert.assertEquals(manifest.getDependencies().get(0).getMetadata().getVersion(), "1.0.5");
+            Assert.assertEquals(manifest.getDependencies().get(1).getModuleID(), "jquery");
+            Assert.assertEquals(manifest.getDependencies().get(1).getMetadata().getVersion(), "2.2.3");
+
+            Files.delete(baloPath);
+            Files.delete(tmpDir);
+        }
+    }
+
+    // Enabled only on Windows
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java:218]: Shall we change the `description` of the test method appropriately and remove this comment?

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java (hunk 2 above)
Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: yes, modified the descriptions accordingly.

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java (hunk 1 above)
Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: refactored

File: compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java (hunk 3 above)
Comment (parkavi11) [reply compiler/ballerina-lang/src/test/java/org/ballerinalang/toml/ManifestProcessorTest.java]: removed

**Pull request closed without merging**